src/Orleans.Core/Messaging/Message.cs

```diff
@@ -1253,6 +1284,11 @@ public static void Serializer(object untypedInput, ISerializationContext context
                     WriteObj(context, typeof(GuidId), input.TargetObserverId);
                 }
 
+                if ((headers & Headers.ACTIVITY_ID) != Headers.NONE)
+                {
+                    WriteObj(context, typeof(GuidId), input.ActivityId);
+                }
```

Comment:
Should this be the last field written since we just added it with the last flag (1 << 30)? I think that may be required for back compat